### ISSUE TITLE
A default build script should support *not* starting with dot

### ISSUE THREAD

**Issue by lzybkr** [nightroman/Invoke-Build#37]
It's a little weird to use the dotfile convention for a default build script. On Windows, it looks weird, on *nix, the file is hidden from `ls` without an option.

I'd suggest allow at least one other default filename as a default. One obvious possibility - if the script is in `foo\bar\project`, then `project.build.ps1` is considered a default build script. In other words, if the filename is `directoryname.build.ps1`, it's the default build script.

**Comment by nightroman** [comment]
@lzybkr, are you aware that *project.build.ps1* IS the default build script if it is the only like `*.build.ps1*?

**Comment by nightroman** [comment]
`.build.ps1` takes over only in the case of 2+ `*.build.ps1` files in the same
directory. This case is so rare that I think we should not really care of it.
The only known example is the directory with IB [Tests](https://github.com/nightroman/Invoke-Build/tree/master/Tests).

The suggested `directoryname.build.ps1` looks promising but it is not "portable".
Renaming a directory (external environment change) may break things inside,
this is not quite right.

In other words, I do not think any change is needed. Except perhaps improving
the manuals or examples, so that users do not think that `.build.ps1` is the
only possible default script. What do you think?

**Comment by lzybkr** [comment]
Sure, documentation could be improved.  I see plenty of examples using `.build.ps1` in https://github.com/nightroman/Invoke-Build/wiki/Build-Scripts-in-Projects

It does seem like you should active discourage using that filename though - it will only be confusing on Linux.

**Comment by nightroman** [comment]
Yes, nowadays `<project-name>.build.ps1` should be recommended as a build
script name, not the shortest possible and *ultimate default* `.build.ps1`.

`.build.ps1` was never *recommended*, if you read the help and wiki. It just
happened that I used this name in my projects. You find this name weird.
I find it convenient for quite a few reasons. "Default" is a convention.
There will be always folks who like and dislike your choice.

`<project-name>.build.ps1` is the default build script right away, this
was supported from the very beginning, with a caveat: it should be the
only build script in its directory. This is almost always the case.

I reviewed the help and wiki and tweaked a little bit.
I added the explicit first line comment to [Build-Scripts-in-Projects](https://github.com/nightroman/Invoke-Build/wiki/Build-Scripts-in-Projects).

I do not think it will be really needed but in the future we may reconsider the
*ultimate default* which is currently `.build.ps1`. In fact, there will be use
cases when having it as "hidden" is actually a convenience. For example, the
Plaster default build scripts is not supposed to modified by users.
